### PR TITLE
Wait for deploy

### PR DESCRIPTION
**Pull request by Vektah**
In previous versions of ecs-cli deploy would poll for a "reached a steady state" in the event log and `deployment.DesiredCount == deployment.RunningCount`. I think this broken for a couple of reasons:
 - its only checking deployment[0], this could actually be the old deployment.
 - It dosen't seem to wait for old tasks to stop
 - If the message ever changes deploys will hang

This PR changes the logic to be `len(deployment) == 0 && deployment[0].task == thisTask`. This means:
 - As soon as all other tasks have finished the deployment is complete. 
 - It isn't based on event messaging
 - When deploy is complete you can hit your app with faith that the code you just deployed is going to serve the request.

Additionally, it fixes a bug that was preventing the events from being displayed (log.Println vs ui.Println)

### CODE REVIEW

File: ecs.go [hunk 1]
```diff
@@ -61,57 +62,49 @@ func (d *dockerImageName) String() string {
...
 
-	return nil
-}
-
-func deploymentById(id string, deployments []*ecs.Deployment) *ecs.Deployment {
-	for _, d := range deployments {
-		if id == *d.Id {
-			return d
-		}
+		time.Sleep(2 * time.Second)
```
Comment (lox) [inline ecs.go]: Is this needed?

Comment (Vektah) [inline ecs.go]: Only to avoid hammering aws. It could probably be lower but these deploys take way longer than 2s being noticeable.

**Comment by lox** [comment]
Aside from the sleep, looks good. What was the previous behaviour and why was it broken?

File: ecs.go (hunk 1 above)
Comment (lox) [inline ecs.go]: Should be pulled out as a constant and probably not in what looks like a getter method.

**Comment by Vektah** [comment]
I think the crux of it was that It used to poll for the `reached a steady state.` string which happens when all of the new apps are up, but dosent wait for the old ones to stop.

It also did some weird things expecting `service.Deployments[0]` to initially be the thing being deployed (which it may or may not, but it seems irrelevant). Now it just looks for the task arn which is passed in.

File: ecs.go (hunk 1 above)
Comment (Vektah) [inline ecs.go]: The diff is a bit wonky, but its at the end of a loop in `PollUntilTaskDeployed`.

**Comment by lox** [comment]
What's the issue with old apps being up?

**Comment by Vektah** [comment]
Deploy isn't finished until I can hit a page be sure it has been served by the new code.

**Comment by lox** [comment]
👍

**Pull request merged**